### ISSUE TITLE
Language detection also looks in the node_modules directory

### ISSUE THREAD

**Issue by alkwa-msft** [microsoft/Oryx#1066]
### Bug Report
I was building a NodeJS function app and oryx was failing since it was not able to properly detect the language runtime. I clearly had a package.json and a yarn.lock file in my project however it didn't detect node. After I stopped trying to yarn install in an earlier step in my CI pipeline, oryx was able to properly detect that the function app was a node function app.

- Where did you experience the problem? E.g. Azure Web Apps, Azure Functions, Azure Container Registry, or offline use.
I experienced this while building an Azure Static Web app. I got help from the Azure Static Web App team and we debugged that the issue was in how oryx was handling the build. (language detection getting confused by pre-existing node_modules directory)

- If your repo is publicly available please share its URL:
No

- What happened?
It did not build my  function app because it couldn't detect a valid language

- What did you expect or want to happen?
I expected it to build correctly.

- How can we reproduce it?
Not exactly sure since I can't share my repo. However by excluding the node_modules directory in language detection it should solve the problem.

- Do you have log output? Please include between the backticks:

```
---End of Oryx build logs---
The function language detected is unsupported or invalid. The following languages are valid: node, dotnet, python. If you are not using Azure Functions, you can skip this check by removing the 'api_location' input from the GitHub workflow file.

```

**Comment by arroyc** [comment]
curious: how is this pre-existing node_modules getting generated? are you deploying them as well or they were  just there due to corrupted environment? if you cannot share the repo can you please share your ci workflow in detail?

**Comment by alkwa-msft** [comment]
I was setting up Ci for my azure static web app. I didn't realize the actual deployment of the azure static web app to azure would also pull in dependencies, build, and then deploy.

So I was effectively pulling in my server dependencies twice (once manually and once through the azure static web app task)

I was working with Anthony (PM on the azure static web app) team and we were trying to figure out how the we're failing to detect the function language runtime. 

I also noticed that for whatever reason despite having two nodejs functions, the task was detecting I needed the PHP runtime and the NodeJS runtime for my azure functions.

At the suggestion of Anthony, I stopped manually getting dependencies and only got the dependencies through the azure static web app task and things started working again. This means the delta was me pulling in dependencies for my azure functions before the task. Which means the only thing that changed was the node_modules directory.
